btcec/schnorr/musig2/nonces.go

```diff
@@ -270,6 +270,16 @@ func GenNonces(options ...NonceGenOption) (*Nonces, error) {
 		return nil, err
 	}
 
+	// If the options contain a secret key, we XOR it with with the tagged
+	// random bytes.
+	if len(opts.secretKey) != 0 {
```

Comment:
Check the length being 32 exactly instead of just not 0? Otherwise we could panic below.

Reply:
Should we throw an error if the secret key is not 0, but also not 32?

Reply:
Looks like the latest diff changes the exact length 👍